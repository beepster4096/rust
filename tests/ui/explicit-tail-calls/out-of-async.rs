// edition: 2021
#![feature(explicit_tail_calls)]

async fn a() {
    become b();
    //~^ error: mismatched function ABIs
    //~| error: mismatched signatures
}

fn b() {}

fn block() -> impl std::future::Future<Output = ()> {
    async {
        become b();
        //~^ error: mismatched function ABIs
        //~| error: mismatched signatures
    }
}

fn main() {}
