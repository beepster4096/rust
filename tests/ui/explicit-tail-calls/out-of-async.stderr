error: mismatched function ABIs
  --> $DIR/out-of-async.rs:5:5
   |
LL |     become b();
   |     ^^^^^^^^^^
   |
   = note: `become` requires caller and callee to have the same ABI
   = note: caller ABI is `"rust-call"`, while callee ABI is `"Rust"`

error: mismatched signatures
  --> $DIR/out-of-async.rs:5:5
   |
LL |     become b();
   |     ^^^^^^^^^^
   |
   = note: `become` requires caller and callee to have matching signatures
   = note: caller signature: `extern "rust-call" fn(ResumeTy)`
   = note: callee signature: `fn()`

error: mismatched function ABIs
  --> $DIR/out-of-async.rs:14:9
   |
LL |         become b();
   |         ^^^^^^^^^^
   |
   = note: `become` requires caller and callee to have the same ABI
   = note: caller ABI is `"rust-call"`, while callee ABI is `"Rust"`

error: mismatched signatures
  --> $DIR/out-of-async.rs:14:9
   |
LL |         become b();
   |         ^^^^^^^^^^
   |
   = note: `become` requires caller and callee to have matching signatures
   = note: caller signature: `extern "rust-call" fn(ResumeTy)`
   = note: callee signature: `fn()`

error: aborting due to 4 previous errors

